### PR TITLE
[converter] More test cases for unknown references

### LINKED ISSUE #1199: New tf converter drops unknown variables
### What happened?

When converting a terraform example using the new (experimental) converter, it drops unknown variables.

### Expected Behavior

It should not drop unknown variables

### Steps to reproduce

- Install the snowflake plugin:`pulumi plugin install resource snowflake`
- Use this example from the [official docs](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/failover_group)
- Run `PULUMI_DEV=1 pulumi convert --from terraform --language pcl --out pcl`

See the follwoing PCL generated 
```
resource "db" "snowflake:index/database:Database" {
  name = "db1"
}
resource "sourceFailoverGroup" "snowflake:index/failoverGroup:FailoverGroup" {
  name                    = "FG1"
  objectTypes             = ["WAREHOUSES", "DATABASES", "INTEGRATIONS", "ROLES"]
  allowedAccounts         = ["<org_name>.<target_account_name1>", "<org_name>.<target_account_name2>"]
  allowedDatabases        = [db.name]
  allowedIntegrationTypes = ["SECURITY INTEGRATIONS"]
  replicationSchedule = {
    cron = {
      expression = "0 0 10-20 * TUE,THU"
      timeZone   = "UTC"
    }
  }
}
resource "targetFailoverGroup" "snowflake:index/failoverGroup:FailoverGroup" {
  name = "FG1"
  fromReplica = {
    organizationName  = "..."
    sourceAccountName = "..."
    name              = fg.name
  }
}

```
Notice here `fg.name` is should be `snowflakeFailoverGroup.fg.name` but that root of the traversal is dropped. 

### Output of `pulumi about`

CLI          
Version      3.69.1-dev.0
Go Version   go1.19.10
Go Compiler  gc

Host     
OS       darwin
Version  13.2.1
Arch     arm64

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already). 

```[tasklist]
### Tasks
```

### PR DESCRIPTION
**Pull request by Zaid-Ajaj**
Closes #1199 as we discussed, it drops the root of the traversal which is correct: the first part of the traversal is a resource type and we drop those when mapping from TF to PCL

### CODE REVIEW

File: pkg/tf2pulumi/convert/testdata/partial_attribute_lookup/main.tf [hunk 1]
```diff
@@ -1,4 +1,21 @@
...
     input_two = unknown_resource_type.example.list[0]
+}
+
+resource "simple_resource" "another_example" {
+    input_one = example_resource_type.resource_name.some_attribute
+    input_two = var.example.test_attribute
+}
+
+output "test_unknown_already_declared_data_source" {
+    value = data.example.another_test_attribute
```
Comment (Frassle) [inline pkg/tf2pulumi/convert/testdata/partial_attribute_lookup/main.tf:12]:
```suggestion
    value = data.unknown_data_source.example.another_test_attribute
```
to check that we use the same var as in line 2

File: pkg/tf2pulumi/convert/testdata/partial_attribute_lookup/main.tf [hunk 2]
```diff
@@ -1,4 +1,21 @@
 resource "simple_resource" "example" {
     input_one = data.unknown_data_source.example.attr
     input_two = unknown_resource_type.example.list[0]
+}
+
+resource "simple_resource" "another_example" {
+    input_one = example_resource_type.resource_name.some_attribute
+    input_two = var.example.test_attribute
```
[the remark this answers is not in the document]
Comment (Zaid-Ajaj) [reply pkg/tf2pulumi/convert/testdata/partial_attribute_lookup/main.tf:8]: I thought we wanted them to clash 😅 alright, will change it

File: pkg/tf2pulumi/convert/testdata/partial_attribute_lookup/main.tf [hunk 3]
```diff
@@ -1,4 +1,21 @@
...
+
+output "test_unknown_already_declared_data_source" {
+    value = data.example.another_test_attribute
+}
+
+output "test_unknown_local_variable" {
+    value = local.some_variable_name
+}
+
+output "test_unknown_already_declared_local_variable" {
```
[the remark this answers is not in the document]
Comment (Zaid-Ajaj) [reply pkg/tf2pulumi/convert/testdata/partial_attribute_lookup/main.tf]: Done ✅

File: pkg/tf2pulumi/convert/testdata/partial_attribute_lookup/main.tf (hunk 1 above)
Comment (Zaid-Ajaj) [reply pkg/tf2pulumi/convert/testdata/partial_attribute_lookup/main.tf]: Done ✅

File: pkg/tf2pulumi/convert/testdata/partial_attribute_lookup/main.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (Zaid-Ajaj) [reply pkg/tf2pulumi/convert/testdata/partial_attribute_lookup/main.tf]: Done ✅

[the remark this answers is not in the document]
Comment (Frassle) [reply pkg/tf2pulumi/convert/testdata/partial_attribute_lookup/main.tf]: Well maybe we want another where it does clash? But I think we want to be sure that vars and locals just get their normal name if there are no clashes.

File: pkg/tf2pulumi/convert/testdata/partial_attribute_lookup/experimental_pcl/main.pp [hunk 4]
```diff
@@ -2,3 +2,16 @@
...
+  inputTwo = someConfig.testAttribute
+}
+output "testUnknownAlreadyDeclaredDataSource" {
+  value = exampledataunknownDataSource.anotherTestAttribute
+}
+output "testUnknownLocalVariable" {
+  value = someVariableName
+}
+output "testUnknownConflictingLocalVariable" {
+  value = theirexample.anotherTestAttribute
```
Comment (Frassle) [inline pkg/tf2pulumi/convert/testdata/partial_attribute_lookup/experimental_pcl/main.pp:16]: huh nice catch here. This should be `theirExample` I think that's a bug in the uniqifier when adding prefixes. Could you fix that up?

**Comment by iwahbe** [comment]
I'm closing since there doesn't seem to be desire to merge. If you would like to merge, feel free to re-open this PR.

**Pull request closed without merging**